internal/inventory/gcpfetcher/fetcher_assets.go

```diff
@@ -0,0 +1,100 @@
+// specific language governing permissions and limitations
+// under the License.
+
+package gcpfetcher
+
+import (
+	"context"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/internal/inventory"
+	gcpinventory "github.com/elastic/cloudbeat/internal/resources/providers/gcplib/inventory"
+)
+
+type (
+	assetsInventory struct {
+		logger   *logp.Logger
+		provider inventoryProvider
+	}
+	inventoryProvider interface {
+		ListAllAssetTypesByName(ctx context.Context, assets []string) ([]*gcpinventory.ExtendedGcpAsset, error)
+	}
+	ResourcesClassification struct {
+		assetType      string
+		classification inventory.AssetClassification
+	}
+)
+
+var ResourcesToFetch = []ResourcesClassification{
+	{gcpinventory.CrmOrgAssetType, inventory.AssetClassificationGcpOrganization},
+	{gcpinventory.CrmFolderAssetType, inventory.AssetClassificationGcpFolder},
+	{gcpinventory.CrmProjectAssetType, inventory.AssetClassificationGcpProject},
+	{gcpinventory.ComputeInstanceAssetType, inventory.AssetClassificationGcpInstance},
+	{gcpinventory.ComputeFirewallAssetType, inventory.AssetClassificationGcpFirewall},
+	{gcpinventory.StorageBucketAssetType, inventory.AssetClassificationGcpBucket},
+	{gcpinventory.ComputeSubnetworkAssetType, inventory.AssetClassificationGcpSubnet},
+	{gcpinventory.IamServiceAccountAssetType, inventory.AssetClassificationGcpServiceAccount},
+	{gcpinventory.IamServiceAccountKeyAssetType, inventory.AssetClassificationGcpServiceAccountKey},
+}
+
+func newAssetsInventoryFetcher(logger *logp.Logger, provider inventoryProvider) inventory.AssetFetcher {
+	return &assetsInventory{
+		logger:   logger,
+		provider: provider,
+	}
+}
+
+func (f *assetsInventory) Fetch(ctx context.Context, assetChan chan<- inventory.AssetEvent) {
+	for _, r := range ResourcesToFetch {
+		f.fetch(ctx, assetChan, r.assetType, r.classification)
+	}
+}
+
+func (f *assetsInventory) fetch(ctx context.Context, assetChan chan<- inventory.AssetEvent, assetType string, classification inventory.AssetClassification) {
+	f.logger.Infof("Fetching %s", assetType)
+	defer f.logger.Infof("Fetching %s - Finished", assetType)
+
+	gcpAssets, err := f.provider.ListAllAssetTypesByName(ctx, []string{assetType})
+	if err != nil {
+		f.logger.Errorf("Could not fetch %s: %v", assetType, err)
+		return
+	}
+
+	for _, item := range gcpAssets {
+		assetChan <- inventory.NewAssetEvent(
+			classification,
+			[]string{item.Name},
+			item.Name,
+			inventory.WithRawAsset(item),
+			inventory.WithCloud(inventory.AssetCloud{
+				Provider: inventory.GcpCloudProvider,
+				Account: inventory.AssetCloudAccount{
+					Id:   item.CloudAccount.AccountId,
+					Name: item.CloudAccount.AccountName,
+				},
+				Organization: inventory.AssetCloudOrganization{
+					Id:   item.CloudAccount.OrganisationId,
+					Name: item.CloudAccount.OrganizationName,
+				},
+				Service: &inventory.AssetCloudService{
+					Name: assetType,
+				},
```

Comment:
using the asset type correctly indicates which service was used, but
1 - it's repetitive, we're already sending that string
2 - format is not human friendly 

a better solution would be using [`Resource.DiscoveryName`](https://cloud.google.com/asset-inventory/docs/reference/rest/v1/Asset#Resource:~:text=as%20Cloud%20Bigtable.-,discoveryName,-string)

but the docs say sometimes this field will be missing. so i want to test this some more and change it in a future PR.